apps/src/code-studio/pd/international_optin/InternationalOptin.jsx

```diff
@@ -0,0 +1,248 @@
+import React, {PropTypes} from 'react';
+import FormController from '../form_components/FormController';
+import FormComponent from '../form_components/FormComponent';
+import DatePicker from '../workshop_dashboard/components/date_picker';
+import moment from 'moment';
+import {DATE_FORMAT} from '../workshop_dashboard/workshopConstants';
+import {
+  Row,
+  Col,
+  ControlLabel,
+  FormGroup
+} from 'react-bootstrap';
+
+export default class InternationalOptin extends FormController {
+  static propTypes = {
+    accountEmail: PropTypes.string.isRequired,
+    labels: PropTypes.object.isRequired
+  };
+
+  /**
+   * @override
+   */
+  onSuccessfulSubmit(data) {
+    window.location = `/pd/international_optin/${data.id}/thanks`;
+  }
+
+  /**
+   * @override
+   */
+  serializeFormData() {
+    const formData = super.serializeFormData();
+    //Object.assign(formData['form_data'], this.getDistrictData());
+
+    return formData;
+  }
+
+  /**
+   * @override
+   */
+  getPageComponents() {
+    return [
+      InternationalOptinComponent
+    ];
+  }
+
+  /**
+   * @override
+   */
+  getPageProps() {
+    return {
+      ...super.getPageProps(),
+      accountEmail: this.props.accountEmail,
+      labels: this.props.labels
+    };
+  }
+}
+
+
+class InternationalOptinComponent extends FormComponent {
+  static propTypes = {
+    accountEmail: PropTypes.string.isRequired
+  };
+
+  handleDateChange = (date) => {
+    // Don't allow null. If the date is cleared, default again to today.
+    date = date || moment();
+    super.handleChange({date: date.format(DATE_FORMAT)});
+  };
+
+  render() {
+    const labels = this.props.labels;
+    const date = (this.props.data && this.props.data.date) ?
+      moment(this.props.data.date, DATE_FORMAT) : moment();
+    // todo: add validation
+
+    return (
+      <FormGroup>
+        {
+          this.buildFieldGroup({
+            name: 'firstName',
+            label: labels.firstName,
+            type: 'text',
+            required: true
+          })
+        }
+        {
+          this.buildFieldGroup({
+            name: 'firstNamePreferred',
+            label: labels.firstNamePreferred,
+            type: 'text',
+            required: false
+          })
+        }
+        {
+          this.buildFieldGroup({
+            name: 'lastName',
+            label: labels.lastName,
+            type: 'text',
+            required: true
+          })
+        }
+        {
+          this.buildFieldGroup({
+            name: 'email',
+            label: labels.email,
+            type: 'text',
+            value: this.props.accountEmail,
```

Comment:
Thinking about this a bit - there are some users that would fill out this form that do not have an email (they mistakenly created a student account way back when). Maybe make this readOnly only if we have an email?

Reply:
Good point.  We will actually route them to a separate page directing them to edit their profile and add an email, similar to what we do for non-teachers and signed-out users.  So anybody who actually gets to this form will have an email.